h2o-r/h2o-package/R/models.R

```diff
@@ -1857,15 +1860,43 @@ h2o.hit_ratio_table <- function(object, train=FALSE, valid=FALSE, xval=FALSE) {
 #' h2o.F1(perf)
 #' }
 #' @export
-h2o.metric <- function(object, thresholds, metric) {
-  if(!is(object, "H2OModelMetrics")) stop(paste0("No ", metric, " for ",class(object)," .Should be a H2OModelMetrics object!"))
-  if(is(object, "H2OBinomialMetrics")){
-    if(!missing(thresholds)) lapply(thresholds, function(t,object,metric) h2o.find_row_by_threshold(object, t)[, metric], object, metric)
-    else {
-     if(missing(metric)) object@metrics$thresholds_and_metric_scores else object@metrics$thresholds_and_metric_scores[, c("threshold", metric)]
+h2o.metric <- function(object, thresholds, metric, transform=NULL) {
```

Comment:
@ledell I had to fix and extend this function to support various cases:
- fix the `error` metrics: one was completely broken, the other was returning confusing (and actually incorrect results): this is fixed here by the generic `transform` param.
- support for "threshold maximizing the metric" like in Python: this is added here by allowing a new special value `thresholds='max'`